### ISSUE TITLE
publish 18 components on the npm

### ISSUE THREAD

**Issue by JosefJezek** [material-components/material-components-web-components#281]
18 components are private

https://github.com/material-components/material-components-web-components/search?q=%22%5C%22private%5C%22%3A+true%22&unscoped_q=%22%5C%22private%5C%22%3A+true%22

**Comment by PaulHMason** [comment]
On a related note - there's been no real feedback on the progress/timelines of the Material Web Components. I know it's still early days but it would be nice to have an idea about what's going on - the whole project appears dead to me.

**Comment by caliny97** [comment]
@Project maintainers - great work, but please stop ignoring requests for project status and timelines.
Just sent an email to polymer-dev@googlegroups.com to see if we "the people" can get an update.  Bcc'ed Sundar Pichai, maybe some connections might help move this project faster.

**Comment by jothepro** [comment]
I don't think that spamming the maintainers is a way to solve the problem.
But I agree, it would be really helpful to get some feedback!!!

**Comment by caliny97** [comment]
Thanks for the reply,
How do you think would be the proper way of gettings some sort of response
from maintainers? When multiple requests from me and other people are being
ignored.
Regards,

**Comment by rwestlund** [comment]
I agree with @caliny97, not sure what else to do. These elements have been "in progress" for over a year now, with few signs of life and no real communication from the devs. I didn't see any mention of these Material web components or lit-html at Google I/O, which was surprising after the prominent role that Polymer played in previous years.

This really looks like another abandoned Google project, which is a shame because I think lit-element is better than React but it's not really usable without these library components. If it is abandoned, it'd be really great if someone over there could tell us so that we can make other plans. Some of us have big apps in production that are still dragging around Polymer 1 compatibility stuff from using paper-elements. We've already rewritten our custom elements on top of lit-element, but now there's growing pressure to rewrite again and move into the React ecosystem (I keep telling them to wait a little longer). We made a bet that the Polymer team would stick around, and we're starting to regret it :(

I really hope that this is a temporary organizational issue and that the team will return to the great work they've been doing to support PWAs. But we just can't sit around and wait for another year.

**Comment by jothepro** [comment]
> Thanks for the reply, How do you think would be the proper way of gettings some sort of response from maintainers? When multiple requests from me and other people are being ignored. Regards,
> […](#)

@caliny97 I guess the only solution would be to fork their project and continue working on it under a rebrand. But I guess this would require at least the permission of the maintainers... _(Disclaimer: I have NO experience in doing stuff like this AT ALL)_

@rwestlund Realistically speaking there is just 1000 people that have star-ed this project, 100 followers and just ~5 people that have actively contributed a bigger amount of code. Thats pretty disappointing numbers for a project that is supposed to lead the web into a brighter future.
I wouldn't be too surprised if google tries to kill this project silently, and hope nobody sees it...

Can anyone give advice if and how a fork with rebranding could be done legally? I'm not too deep into the tech but since polymer 3 heavily relies on web-standards that are already implemented by all major browser (like the[ template-tag](https://www.w3schools.com/TagS/tag_template.asp) and [shadow dom](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM) ), a[ disaster like polymer (1)](https://www.neowin.net/news/mozilla-executive-claims-that-google-has-made-youtube-slower-on-edge-and-firefox) is not to be expected.


But maybe the maintainers are really just on holiday and will be back soon, even more motivated than before... :palm_tree: :stuck_out_tongue_winking_eye:

**Comment by PaulHMason** [comment]
I suppose a new project built on top of [Material Components for the Web](https://github.com/material-components/material-components-web/) is one option (i.e. don't fork and redo from scratch), or just use the new [Ionic Stencil based components](https://ionicframework.com/docs/components).

**Comment by rwestlund** [comment]
@jothepro the code here is Apache 2.0 and the CSS repo it's based on is under MIT, so there are no legal obstacles to a fork. I doubt anyone here has time for that though.

@PaulHMason The Ionic components looked promising at first, but I can't get them to work without buying into their whole framework, despite what they say about portability. Their example for non-framework usage is not an ES6 module, and requires global stylesheets. I tried importing just one component, but it didn't extend HTMLElement and didn't register itself :/

**Comment by johnthad** [comment]
@rwestlund This project was mentioned on the Slack channel yesterday: https://weightless.dev/

**Comment by celesteking** [comment]
Sudar told on conference that Google cancelled the project - they're gonna use Ember.js from now on. It was a difficult decision , but they decided to concentrate their resources on android 10 and waymo . Material design is going away, too, as nobody liked it anyway.

**Comment by thexs-dev** [comment]
@celesteking does that mean that LitElement and lit-html are dead too?

I am still stuck with Polymer 1 and was (still am?) quite excited with migrating to new the Lit concept

**Comment by jsilvermist** [comment]
Please don't feed the trolls, they bite.

**Comment by caliny97** [comment]
@johnthad - thanks for the link. I just tried http://weightless.dev for the last 4 hours, My way of trying a new framework is by building a simple CRUD that looks good and incorporates most component people are used to see in a web app, If i can build it in less than 3 hours then it's a pass. Weightless might be a good candidate for people looking for UI components based on lit-element. Yup you heard it right.... I stepped thru the code and was so exited to see "extends LitElement".

**Comment by iantrich** [comment]
@caliny97 I did the same the other day and am liking what I'm seeing so far. I wish I could figure out how to use ionic's as well, but that didn't work out for me, as I like the look of their elements a bit more. 🤷‍♂

**Comment by caliny97** [comment]
@iantrich - I tried ionic two months ago and managed to get it to work with lit-element, the components are nicely done and follow material guides lines very close but I felt it tries lock you into their whole ionic framework.

**Comment by iantrich** [comment]
@caliny97 you wouldn't happen to still have that test around in a gist, would you?

**Comment by caliny97** [comment]
@iantrich - I worked with ionic while doing a favor of getting a POC running when an outsourced developer ghosted. I can't share the code but the key is to use fat-arrows to wire events, I think that's where most people have issues with. `<ion-button (click)='${(e:any) => this._handleClick(e)}'></ion-button>`

**Comment by rwestlund** [comment]
@johnthad Thanks, weightless looks promising. It needs some polish, but he's done more since March than the devs here have done in a year. I'm still playing with it, but it looks good so far -- nice and light weight, close to the real native elements, and built on lit!

**Comment by libertizer** [comment]
@celesteking  At which conference Sudar announed this? Do you have the link? They should hand over google material web to the community if they don't want to continue development...

**Comment by PaulHMason** [comment]
> @celesteking At which conference Sudar announed this? Do you have the link? They should hand over google material web to the community if they don't want to continue development...

He's just trolling - nothing was announced.

**Comment by celesteking** [comment]
Now I realize how fake news stations like Fox/RT manage to spread lies so easily. You have to plant a tiny seed of discord, distract, distort and in no time you're ready to harvest.
***

My **real complaint** was that Google deprecated polymer v3 before actually releasing the newer version, the one that supposed to overtake it. That's definitely a big mismanagement on their part. 

I'm a newbie looking for bootstrap/vue.js/whatever replacement, looking for options. I have stumbled upon this polymerjs project because someone said something on reddit/internets and I followed the link. Next, I follow documentation links, go to API/Guide only to see a deprecation notice "blah blah you better use our newer (LINK)". I click on that link and end up in this half assed repo.  

That's not how you do things. You should first release something that's usable, and only after that you can start deprecating the old one.

__Take this very repo readme__:

### Quick start
The easiest way to try out the Material Web Components is to use one of these online tools:
Runs in all supported browsers: [Glitch](https://glitch.com/edit/#!/material-web-components)

Click on glitch link, get an error. Click on https://glitch.com/edit/#!/mwc-icon-example?path=index.html:1:0 demo, get an error. 

As a newbie, do I want to mess further with this non-working stuff? No.

**Comment by danielgek** [comment]
@celesteking they have a successor of Polimer 3 is LitElement (https://lit-element.polymer-project.org/), tho i agree with you about the repo :/

**Comment by libertizer** [comment]
It is however no joke that Kenneth G. Franqueiro, has left google, he was the lead developer for google material web. I wonder what this means for the project. I'm actually developing a real project in Google Material and I like it a lot :-) The only missing part for me is the datatable...

**Comment by Sebastianhex** [comment]
> It is however no joke that Kenneth G. Franqueiro, has left google, he was the lead developer for google material web. I wonder what this means for the project. I'm actually developing a real project in Google Material and I like it a lot :-) The only missing part for me is the datatable...

Even with the apparent loss of the lead developer for Google Material Web, I'm remaining hopeful that they'll be continuing with the project soon, my team is looking forward to adding more MWC to our project! Especially the [Dialog](https://material.io/design/components/dialogs.html), [Menu](https://material.io/design/components/menus.html) and [Snackbar](https://material.io/design/components/snackbars.html) components.
I think Google has come too far with [Polymer](https://www.polymer-project.org/) and [Material Components](https://github.com/material-components) to not finish up on their [Material Web Components](https://github.com/material-components/material-components-web-components).

Eagerly awaiting to see progress and more published components soon!

**Comment by caliny97** [comment]
> > It is however no joke that Kenneth G. Franqueiro, has left google, he was the lead developer for google material web. I wonder what this means for the project. I'm actually developing a real project in Google Material and I like it a lot :-) The only missing part for me is the datatable...
> 
> Even with the apparent loss of the lead developer for Google Material Web, I'm remaining hopeful that they'll be continuing with the project soon, my team is looking forward to adding more MWC to our project! Especially the [Dialog](https://material.io/design/components/dialogs.html), [Menu](https://material.io/design/components/menus.html) and [Snackbar](https://material.io/design/components/snackbars.html) components.
> I think Google has come too far with [Polymer](https://www.polymer-project.org/) and [Material Components](https://github.com/material-components) to not finish up on their [Material Web Components](https://github.com/material-components/material-components-web-components).
> 
> Eagerly awaiting to see progress and more published components soon!

I don't think Polymer/LitElement is going away anytime soon, Goggle invested a lot into it. YouTube was redesigned with Polymer 3 and there are now some part in lit. I think is just this project being neglected by maintainers. It's really sad that after all the noise from this thread, we haven't heard a word from them.  I am moving forward with https://weightless.dev for a big project I am starting.

**Comment by PaulHMason** [comment]
None of this detracts from the value of LitElement and lit-html, it would just be nice to know what's happening with the Material Web Components projects. Other libraries and frameworks (React and Vue come to mind) don't have an official UI component suite and 3rd parties have stepped in to fill the gap (Vue has some excellent implementations). I'm just a bit wary of committing to something like weightless because it's supported by a single developer and doesn't have a track record (chicken/egg thing). 

It's tempting to develop my own component suite, but it's not a trivial undertaking.

**Comment by claytongulick** [comment]
Thanks for the pointer to https://weightless.dev it looks like exactly what I'm after. Seems to be missing a drawer / slide out menu component though?

**Comment by PaulHMason** [comment]
> @iantrich by developing your own component suite, you've effectively created another project that has a single developer; something that you are wary of. Why not become a contributor to weightless instead? Just curious.

Good point, with a long answer... Mainly because it's just something I'm interested in doing, but also because there's isn't really an "optimized" component suite for building line of business applications. Most commercial products like those from Telerik, DevExpress, etc. are huge kitchen sink suites that are overkill for most internal CRUD type applications. Even supporting the full Material Design spec (V2) is a bit much. I just want something with as few knobs to turn as possible to get the job done.

**Comment by yorrd** [comment]
@PaulHMason then again, do some tree shaking and it doesn't really matter anymore how big the library is right?

**Comment by thexs-dev** [comment]
Quick but important question from my side:
+ What would I miss and gain, if while waiting (or instead of) for this project to complete, I start using Vue and Vuetify, to migrate some projects out of Polymer 1?

Thanks for your feedback

**Comment by lkraav** [comment]
One of the top custom element component suites https://vaadin.com/components are roadmapping their Polymer -> LitElement migration hopefully to happen within this year.

**Comment by PaulHMason** [comment]
@thexs-dev The problem is nobody has any idea when the project will be complete. There's nothing wrong with Vue and Vuetify (or https://vuematerial.io/), but then again weightless is pretty solid if it meets your needs.

**Comment by caliny97** [comment]
> @thexs-dev The problem is nobody has any idea when the project will be complete. There's nothing wrong with Vue and Vuetify (or https://vuematerial.io/), but then again weightless is pretty solid if it meets your needs.

I moved on with https://weightless.dev for a big project I started last week, It really surpassed my expectations for a newcomer library. The code is well written, the api is really light and the documentation is well written.  Vue/Vuetify are also a good choice, I did some big project with those.

@thexs-dev - My project is a complex analytics dashboard, here is the chosen stack if it helps you to make a choice.  Lit-element , weightless , GRPC , vertx

**Comment by eskan** [comment]
Ok so we we have been eared since @azakus update the README to mentioned this project is a way to get MDC to web components and they still work on how they will achieve this.
That's what i like with **weightless**, it's not a wrapping to an other component, and finally i found cool that is not fully material compliant. So like most people here i choose to switch to WL for now and also i encourage people to contribute to andrea's work.

**Comment by rwestlund** [comment]
I'm definitely going with weightless for the time being. But now my ugly-ass app has components from:
- paper
- vaadin
- mwc
- weightless
- some random third-party stuff that wasn't in one of these libraries

It's a little visually jarring, but such is the nature of the ecosystem I suppose. Someday I'll finally cut the Polymer 1 compat stuff that paper and vaadin are pulling in and then maybe Lighthouse and I can be friends again...

FYI @andreasbm, You're becoming popular over here. You might see a spike in contributions.

**Comment by karimayachi** [comment]
So, it seems there's a bit of activity surrounding MWC... Still, thanks for mentioning Weightless. I'm impressed with the amount of work that has been done over there. It's far from 'production ready' though. And it doesn't play nice with Knockout, my weapon of choice.

**Comment by PaulHMason** [comment]
@karimayachi > So, it seems there's a bit of activity surrounding MWC... Still, thanks for mentioning Weightless. I'm impressed with the amount of work that has been done over there. It's far from 'production ready' though. **And it doesn't play nice with Knockout, my weapon of choice**.

You've got it the wrong way around - Knockout doesn't play nicely with Web Components :)  You have a good point though, and I was thinking about Knockout the other day. With the advent of composable web applications using Web Components, Knockout has suddenly become relevant again (as pure data binding library). Maybe raise the issue with the Knockout maintainers. The way I see it is there's 2 issues:

- Being able to use Knockout for authoring Web Components.
- Being able to use Knockout to glue together 3rd party Web Components in an application.

**Comment by karimayachi** [comment]
Hi @PaulHMason ,

Yes, and no... :-)

Yes, Knockout has some conceptual problems when it comes to Web Components (WC) in general. Knockout manipulates the DOM, which can collide with WCs that also manipulate the DOM. Also in the Knockout world bindings are two-way, which LitElement has moved away from.

But these conceptual problems were not what I was referring to. If you have fairly straightforward bindings and straightforward WCs, it should just work.

Weightless (WL) however has additional problems, that seem to be WL specific. Mainly if you have a WL component inside a foreach binding, the component will get corrupted and doubles the HTML elements inside the Light DOM of the WL component. I don't see the same behaviour with the MWC components, so I think this is WL specific. That's what I meant with WL doesn't play nice with Knockout.

I absolutely agree that the widespread use of WCs makes Knockout relevant again. Especially to glue together 3rd party WCs. I think LitElement got the authoring part covered.

It would be great to pull in 3rd party components as blackbox building blocks and use an unopinionated library as Knockout to bind everything together, instead of wrapping the whole thing in a component based, template based, state store based monstrosity... (you know who you are ;-) )

I will take this discussion over to the Knockout GitHub or Google Group

**Comment by JosefJezek** [comment]
Fixed https://github.com/material-components/material-components-web-components/commit/eca942003db9a1d1cada85192d629aeaf6aab737
